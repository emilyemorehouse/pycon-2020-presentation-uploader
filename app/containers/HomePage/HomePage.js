import React from 'react'
import { Helmet } from 'react-helmet'
import { FormattedMessage } from 'react-intl'
import PropTypes from 'prop-types'

// Components
import { Box } from 'components/Box'
import { Button } from 'components/Button'
import { Heading } from 'components/Heading'
import { Text } from 'components/Text'

// Messages
import messages from './HomePage.messages'

/*
 * HomePage
 *
 * The HomePage renders:
 * - The page title and metadata using Helmet
 * - The page header and description
 *
 */
export const HomePage = ({ history }) => (
  <>
    <Helmet>
      <title>Home Page</title>
      <meta name="description" content="A React.js Boilerplate application homepage" />
    </Helmet>

    <Box align="center">
      <Heading level="2" color="brand">
        <FormattedMessage {...messages.startProjectHeader} />
      </Heading>

      <Text>
        <FormattedMessage {...messages.startProjectMessage} />
      </Text>

      <Box gap="small" direction="row" pad="large">
        <Button primary label="Login" onClick={() => history.push('/login')} />
        <Button primary label="Register" onClick={() => history.push('/register')} />
      </Box>
    </Box>
  </>
)

HomePage.propTypes = {
  history: PropTypes.object,
}

export default HomePage
