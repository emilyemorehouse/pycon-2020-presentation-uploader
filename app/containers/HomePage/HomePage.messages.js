import { defineMessages } from 'react-intl'

export const scope = 'boilerplate.containers.HomePage'

/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
export default defineMessages({
  startProjectHeader: {
    id: `${scope}.start_project.header`,
    defaultMessage: 'LiveRider',
  },
  startProjectMessage: {
    id: `${scope}.start_project.message`,
    defaultMessage: 'The link between music venues, touring artists, and crew.',
  },
})
