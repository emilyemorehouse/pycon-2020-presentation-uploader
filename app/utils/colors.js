const colors = {
  yellow: 'rgb(225, 189, 71);',
  crimson: 'rgb(230,34,45)', // Crimson
  magenta: 'rgb(221,17,117)', // Magenta
  modBlue: 'rgb(65,86,166)', // Mod Blue
  cuttleBlue: 'rgb(24,192,221)', // Cuttle Blue
  darkGrey: 'rgb(68,68,68)', // Dark Grey
  mediumGrey: 'rgb(170,170,170)', // Medium Grey
  lightGrey: 'rgb(238,238,238)', // Light Grey
}

colors.primary = colors.modBlue
colors.secondary = colors.yellow
colors.primaryText = colors.darkGrey
colors.background = colors.lightGrey

export default colors
