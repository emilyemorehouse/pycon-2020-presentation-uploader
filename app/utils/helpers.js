import { isPlainObject } from 'lodash'

const ERROR_DEFAULT = 'An error occurred. Please try again.'
const ERROR_NO_INTERNET = 'Request could not be made. Please check your internet connection.'

export const createWithDoc = ({ envName = '', docFunction = () => {}, component = '' }) => {
  let createComponentWithDoc
  if (envName !== 'production') {
    createComponentWithDoc = docFunction(component) // eslint-disable-line global-require
  }
  return createComponentWithDoc || component
}

export const getErrorMessage = err => {
  // - Axios does not return a response object if the service cannot be reached
  if (!err.response) {
    return ERROR_NO_INTERNET
  }

  // - If we do have a response and data object, use the full error object
  if (err.response && err.response.data) {
    try {
      const errorStringOrObject = Object.values(err.response.data)[0]
      if (isPlainObject(errorStringOrObject)) return errorStringOrObject[0]
      return errorStringOrObject
    } catch (error) {
      // eslint-disable-next-line no-console
      console.warn('Could not infer error message from error response object')
    }
  }

  // If a message has not been set, fallback to our default message
  return ERROR_DEFAULT
}
