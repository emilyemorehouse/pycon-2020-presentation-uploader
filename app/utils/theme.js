import colors from './colors'

const theme = {
  global: {
    font: {
      family: 'Roboto Mono',
      size: '16px',
    },
    colors: {
      brand: colors.primary,
      focus: colors.secondary,
    },
  },
  button: {
    border: {
      radius: '4px',
    },
  },
}

export default theme
