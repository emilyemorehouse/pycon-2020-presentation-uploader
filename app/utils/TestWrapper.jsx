/* eslint-disable react/prop-types */
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { IntlProvider } from 'react-intl'
import { Grommet } from 'grommet'

import theme from './theme'

export const TestWrapper = ({ children }) => (
  <IntlProvider locale="en">
    <MemoryRouter initialEntries={['/']}>
      <Grommet theme={theme}>{children}</Grommet>
    </MemoryRouter>
  </IntlProvider>
)
