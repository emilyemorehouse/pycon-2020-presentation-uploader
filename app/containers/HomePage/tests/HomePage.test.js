import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'

// Components
import { TestWrapper } from 'utils/TestWrapper'
import { HomePage } from '../HomePage'

/**
 * HomePage Test Criteria
 *
 */
describe('HomePage', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestWrapper>
        <HomePage loading={false} error={false} repos={[]} />
      </TestWrapper>,
    )

    expect(firstChild).toMatchSnapshot()
  })
})
