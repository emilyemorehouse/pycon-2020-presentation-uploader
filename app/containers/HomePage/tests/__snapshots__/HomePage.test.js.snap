// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render and match the snapshot 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 blnwXx"
>
  <div
    class="StyledBox-sc-13pk1d4-0 gVSbfC"
  >
    <h2
      class="StyledHeading-sc-1rdh4aw-0 hoUPUX"
    >
      <span>
        LiveRider
      </span>
    </h2>
    <span
      class="StyledText-sc-1sadyjn-0 hUokoe Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
    >
      <span>
        The link between music venues, touring artists, and crew.
      </span>
    </span>
    <div
      class="StyledBox-sc-13pk1d4-0 hBDGhc"
    >
      <button
        class="StyledButton-sc-323bzc-0 iNqTxf Button__StyledGrommetButton-sc-5cxguj-0 dKGxUf"
        type="button"
      >
        Login
      </button>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
      />
      <button
        class="StyledButton-sc-323bzc-0 iNqTxf Button__StyledGrommetButton-sc-5cxguj-0 dKGxUf"
        type="button"
      >
        Register
      </button>
    </div>
  </div>
</div>
`;
